core/services/keystore/orm.go

```diff
@@ -0,0 +1,101 @@
+package keystore
+
+import (
+	"math/rand"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/csakey"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/ethkey"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/ocrkey"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/p2pkey"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/vrfkey"
+	"gorm.io/gorm"
+)
+
+func NewORM(db *gorm.DB) ksORM {
+	return ksORM{
+		db: db,
+	}
+}
+
+type ksORM struct {
+	db *gorm.DB
+}
+
+func (orm ksORM) saveEncryptedKeyRing(kr *encryptedKeyRing) error {
+	err := orm.db.Exec(`
+		UPDATE encrypted_key_rings
+		SET encrypted_keys = ?
+		WHERE id = (
+			SELECT id FROM encrypted_key_rings ORDER BY id ASC LIMIT 1
+		)
```

Comment:
How does this query work? It looks like you just always update the first key_ring in the database

Reply:
that's the point! there is only ever 1 key ring